WordPress/WordPressTest/RemoteFeatureFlagTests.swift

```diff
@@ -66,6 +69,54 @@ class RemoteFeatureFlagTests: XCTestCase {
         XCTAssertFalse(store.value(for: MockFeatureFlag.remotelyDisabledLocallyEnabledFeature))
         XCTAssertFalse(store.value(for: MockFeatureFlag.remotelyDisabledLocallyDisabledFeature))
     }
+
+    func testThatUpdateIfNeededDoesNotUpdateIfCacheIsValid() {
+        // Given
+        let userDefaults = UserPersistentStoreFactory.instance()
+        let recentDate = Date(timeInterval: -1, since: Date())
+        userDefaults.set(recentDate, forKey: RemoteFeatureFlagStore.Constants.LastRefreshDateKey)
+        let mock = MockFeatureFlagRemote(flags: MockFeatureFlag.remoteCases)
+        let store = RemoteFeatureFlagStore.shared
+
+        // When
+        store.updateIfNeeded(using: mock)
+
+        // Then
+        // The remote flags should not be present
+        XCTAssertFalse(store.hasValue(for: MockFeatureFlag.remotelyEnabledLocallyEnabledFeature))
+    }
+
+    func testThatUpdateIfNeededUpdatesIfCacheIsExpired() {
+        // Given
+        let userDefaults = UserPersistentStoreFactory.instance()
+        let distantDate = Date(timeInterval: -90_000, since: Date())
```

Comment:
This test could wind up flaky if you hard-code the value in here. It might make more sense to base this on the TTL?

Reply:
The TTL is private, we'll have to expose it just for the sake of the tests.
If the TTL changes the test would not become flaky, it will consistently fail which will prompt us to update the tests. I'm with you that we should always avoid flakiness, but I don't think this case, in particular, constitutes flakiness.

Reply:
Good point. Updated in 5ad909f99e60ca497ef216632517bf4b3f5f9409